src/lib/crypto/kimchi_backend/pasta/pallas_based_plonk.ml

```diff
@@ -34,23 +34,17 @@ end
 module R1CS_constraint_system =
   Kimchi_pasta_constraint_system.Pallas_constraint_system
 
-let lagrange : int -> _ Kimchi_types.poly_comm array =
-  Memo.general ~hashable:Int.hashable (fun domain_log2 ->
-      Array.map
-        Precomputed.Lagrange_precomputations.(
-          pallas.(index_of_domain_log2 domain_log2))
-        ~f:(fun unshifted ->
-          { Kimchi_types.unshifted =
-              Array.map unshifted ~f:(fun (x, y) -> Kimchi_types.Finite (x, y))
-          ; shifted = None
-          } ) )
+let lagrange srs domain_log2 : _ Kimchi_types.poly_comm array =
+  let domain_size = Int.pow 2 domain_log2 in
+  Array.init domain_size ~f:(fun i ->
+      Kimchi_bindings.Protocol.SRS.Fq.lagrange_commitment srs domain_size i )
```

Comment:
confirm: this should be Pallas's scalar field `Fq`?

Reply:
That's right, yea. Happily, the typechecker makes sure that we have the right one anyway :)